static/app/utils/profiling/gl/utils.ts

```diff
@@ -682,3 +682,85 @@ export function computeConfigViewWithStategy(
 
   return frame.withHeight(view.height);
 }
+
+export class RectCursorUtil {
```

Comment:
i made this a standalone class but don't have strong opinions about it, these could all be standalone util functions. Some of the methods aren't actually used, but i filled them out for completeness.

The gist of what this class does is creates two 'buffered' virtual rects in and around the actual rect that are (in this case the mini map view).

The area between the outer rect and the inner rect becomes our resize window area... rect's our border.

The inner area becomes our grab area to drag.


<img width="1013" alt="image" src="https://user-images.githubusercontent.com/7349258/204392512-4d5b94ea-214d-42ee-9597-3f80e0ba876c.png">

Reply:
I think we can avoid creating a new rect for each mouse move and just compute the distance from the edges to the cursor. One way of doing this without doing a bounds check would be to take the min value of the absolute distance between the edges and the cursor. Something like min(abs(left edge - cursor x), abs(right edge - cursor x)) would give you either the left or right edge that you are closest to. Similar with min(abs(top edge - cursor y), abs(bottom edge - cursor y)) would give you top/bottom edge. If you take the min distance of those 4, you get the edge that you are nearest to. 

The benefit of that is that you are being more precise as you always take a min distance (which could be the case when the rect is very small).